homeassistant/components/mqtt/client.py

```diff
@@ -639,6 +746,8 @@ def _mqtt_on_connect(
             async def publish_birth_message(birth_message: PublishMessage) -> None:
                 await self._ha_started.wait()  # Wait for Home Assistant to start
                 await self._discovery_cooldown()  # Wait for MQTT discovery to cool down
+                # Update subscribe cooldown period to a shorter time
+                self._subscribe_debouncer.set_timeout(SUBSCRIBE_COOLDOWN)
```

Comment:
Should we reset to `INITIAL_SUBSCRIBE_COOLDOWN` when disconnected from the broker?